src/parent/parent.js

```diff
@@ -853,8 +863,15 @@ export function parentComponent<P, X, C>({
       })
       .then((isClosed) => {
         if (!cancelled) {
-          if (isClosed) {
-            return close(new Error(`Detected ${context} close`));
+          if (context === CONTEXT.POPUP && isClosed) {
+            return close(new Error(COMPONENT_ERROR.POPUP_CLOSE));
+          } else if (
+            context === CONTEXT.IFRAME &&
+            isClosed &&
+            ((currentContainer && isElementClosed(currentContainer)) ||
+              isSecondRenderFinished)
```

Comment:
where did we land on checking for second render? I thought we said this might cause weird issues but I don't remember if we changed our minds later

Reply:
We did decide to not use the condition of waiting until second render was finished by itself, as that could cause issues when the container is removed for a legitimate reason before it's finished.

Essentially I combined options 1 and 2 and it seemed to work well. If the container is removed, OR second render is finished then we should call `close` here.

Reply:
okay sounds good, thank you!

Reply:
I am a little worried about adding the new isSecondRenderFinished logic here. The prerender phase still renders inside an `<iframe>` with no src tag. If that gets destroyed, we would still want this error to be thrown.